README.md

```diff
@@ -1,48 +1,147 @@
 # terraform-registry-address
 
-This package helps with representation, comparison and parsing of
-Terraform Registry addresses, such as
-`registry.terraform.io/grafana/grafana` or `hashicorp/aws`.
+This module enables parsing, comparison and canonical representation of
+[Terraform Registry](https://registry.terraform.io/) **provider** addresses
+(such as `registry.terraform.io/grafana/grafana` or `hashicorp/aws`)
+and **module** addresses (such as `hashicorp/subnets/cidr`).
 
-The most common source of these addresses outside of Terraform Core
-is JSON representation of state, plan, or schemas as obtained
-via [`hashicorp/terraform-exec`](https://github.com/hashicorp/terraform-exec).
+**Provider** addresses can be found in
 
-## Parsing Provider Addresses
+ - [`terraform show -json <FILE>`](https://www.terraform.io/internals/json-format#configuration-representation) (`full_name`)
+ - [`terraform version -json`](https://www.terraform.io/cli/commands/version#example) (`provider_selections`)
+ - [`terraform providers schema -json`](https://www.terraform.io/cli/commands/providers/schema#providers-schema-representation) (keys of `provider_schemas`)
+ - within `required_providers` block in Terraform configuration (`*.tf`)
 
-### Example
+**Module** addresses can be found within `source` argument
+of `module` block in Terraform configuration (`*.tf`)
+and parts of the address (namespace and name) in the Registry API.
+
+## Compatibility
+
+The module assumes compatibility with Terraform v0.12 and later,
+which have the mentioned JSON output produced by corresponding CLI flags.
+
+We recommend carefully reading the [ambigouous provider addresses](#Ambiguous-Provider-Addresses)
+section below which may impact versions `0.12` and `0.13`.
+
+## Related Libraries
+
+Other libraries which may help with consuming most of the above Terraform
+outputs in automation:
+
+ - [`hashicorp/terraform-exec`](https://github.com/hashicorp/terraform-exec)
+ - [`hashicorp/terraform-json`](https://github.com/hashicorp/terraform-json)
+
+## Usage
```

Comment:
Would it be worth adding [testable examples](https://go.dev/blog/examples) that would render in Go documentation as well?

Reply:
Added.